update-project-version/action.yml

```diff
@@ -0,0 +1,62 @@
+name: Update project version
+description: Update the version of a Python project and create a pull request.
+inputs:
+  project-directory:
+    description: Path to the directory containing pyproject.toml.
+    default: ${{ github.workspace }}
+  branch-prefix:
+    description: The branch prefix to use when creating pull requests.
+    default: users/build/
+  create-pull-request:
+    description: >
+      Specifies whether to create a pull request. Set this to false to update
+      the project version without creating a pull request.
+    default: true
+  version-rule:
+    description: >
+      Specifies the rule for how to update the version number, such as "major",
+      "minor", or "patch". See https://python-poetry.org/docs/cli/#version for
+      the list of rules and their behavior.
+    default: patch
+  use-dev-suffix:
+    description: Specifies whether to use development versions like "1.0.0.dev0".
+    default: true
+runs:
+  using: composite
+  steps:
+  - name: Set variables
+    id: set-vars
+    run: |
+      base_branch=$(git symbolic-ref --short HEAD)
```

Comment:
This doesn't work for a tagged release build.
https://github.com/ni/nitypes-python/actions/runs/15196355990/job/42741408905

```
fatal: ref HEAD is not a symbolic ref
Error: Process completed with exit code 128.
```

Reply:
I implemented a tentative fix based on how we got the branch name in https://github.com/ni/measurement-plugin-python/blob/main/.github/workflows/Publish_NIMS.yml

I guess I need to make a release of *something* before I can test it.

Reply:
Tested with bkeryan/nitypes-python